src/GitHub.Exports/Helpers/ThreadingHelper.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+using ThreadHelper = Microsoft.VisualStudio.Shell.ThreadHelper;
+using GitHub.Extensions;
+
+namespace GitHub.Helpers
+{
+    public static class ThreadingHelper
+    {
+        public async static Task SwitchToMainThreadAsync()
+        {
+            if (!Guard.InUnitTestRunner())
```

Comment:
I'm pretty sure awaiting this will not do the job you want... because this is returning a Task when you await it it's going to try and continue on the capture context and not continue on the UI thread.

To prevent yourself form having to calling `*.ConfiguerAwait(false)` you can do something like this gist for how i've solved this in the past
https://gist.github.com/tocsoft/afc9b5be41722f978ff0414d33afb1f1#file-theadinghelper-cs-L30
and 
https://gist.github.com/tocsoft/afc9b5be41722f978ff0414d33afb1f1#file-theadinghelper-cs-L58